packages/ListBoxBrowser/src/helpers.js

```diff
@@ -0,0 +1,259 @@
+        refDefaultSelectedOptions.current = { ...refDefaultSelectedOptions.current, [option.parent]: [option] };
+      }
+      /* eslint-enable no-param-reassing */
+    }
+
+    // checking if its initial view
+    if (_hasOptions && defaultSelectedView && defaultSelectedView(option.attributes)) {
+      refDefaultSelectedView.current = newPath;
+    }
+
+    return option;
+  });
+}
+
+export function getOptionByKey(data, path) {
+  let node = null;
+  function runner(data, path) {
+    return data.map(option => {
+      if (!option.$$key.includes("/") && (typeof data[path] !== "undefined" && option.$$key === path)) {
+        node = option;
+        throw new Error("option found");
+      }
+
+      if (option.$$key === path) {
+        node = option;
+        throw new Error("option found");
+      }
+
+      if (hasOptions(option.attributes.options)) {
+        return runner(option.attributes.options, path);
+      }
+
+      return [];
+    });
+  }
+
+  try {
+    runner(data, path);
+    return [];
+  } catch (e) {
+    return node;
+  }
+}
+
+export function getBreadcrumb({ data, option }) {
+  let activeOption = option;
+  let key = null;
+  const breadcrumb = [];
+  while (!isRoot(key)) {
+    if (!isRoot(activeOption.parent)) {
+      const parent = getOptionByKey(data, activeOption.parent);
+      breadcrumb.push(parent);
+      activeOption = parent;
+    }
+    key = activeOption.parent;
+  }
+
+  return breadcrumb.reverse();
+}
+
+export function onChange({ source, indexes, list, isParentSelectable, setSelectedOptions, browserKey, isMulti }) {
+  if (!isMulti) {
+    setSelectedOptions(selectedOptions => {
+      const key = isRoot(source) ? "root" : browserKey;
+      if (Object.keys(selectedOptions).length) {
+        const option = getOptions(indexes, list)[0];
+        const isSameOption = Object.keys(selectedOptions).some(key => {
+          return selectedOptions[key][0].$$key === option.$$key;
+        });
+
+        if (isSameOption) {
+          return {};
+        }
+      }
+
+      return { [key]: getOptions(indexes, list) };
+    });
+
+    return;
+  }
+
+  if (isRoot(source)) {
+    const options = getOptions(indexes, list)
+      .map(option => {
+        if (isSelectable({ hasOptions: option.hasOptions, isParentSelectable })) {
+          return null;
+        }
+
+        return option;
+      })
+      .filter(chunk => chunk);
+
+    if (options.length) {
+      setSelectedOptions(selectedOptions => {
+        return { ...selectedOptions, root: [...options] };
+      });
+      return;
+    }
+
+    setSelectedOptions(selectedOptions => {
+      return { ...selectedOptions, root: [] };
+    });
+  }
+
+  setSelectedOptions(selectedOptions => {
+    return { ...selectedOptions, [browserKey]: getOptions(indexes, list) };
+  });
+}
+
+export function focusListBoxBrowser() {
+  window.requestAnimationFrame(() => {
+    document.querySelectorAll('[data-ppk-anchor="listbox-content-inline"]')[1].focus();
```

Comment:
I wonder what if in a future scenario, we have two listbox browsers on the same page. Will there be issues then with this?